projects/angular/src/forms/datepicker/date-input.ts

```diff
@@ -299,4 +301,13 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
       .pipe(filter(date => !!date))
       .subscribe(() => this.datepickerFocusService.focusInput(this.el.nativeElement));
   }
+
+  private retriggerValidation() {
+    if (this.datepickerHasFormControl() && !this.usingNativeDatepicker()) {
+      this.control?.control?.updateValueAndValidity({ onlySelf: true, emitEvent: false }); // Added onlySelf, emitEvent purposefully inorder to prevent unnecessary valuechange triggers
+      if (this.control.control && this.control.control.touched) {
+        this.control.control.setErrors(this.control.control.errors); // Inorder to revalidate and apply validation errors(if any) only when form control is touched.
+      }
+    }
+  }
```

Comment:
```suggestion
 private triggerControlValidation() {
    if (this.datepickerHasFormControl()) {
      this.control.control?.updateValueAndValidity({ onlySelf: true, emitEvent: false });
      this.control.control?.setErrors(this.control.control.errors);
    }
  }
```

- Put the function where the first private functions start (you should receive an error for order probably from the lint)
- `usingNativeDatepicker` is  not needed as validation would happen always when `clrDate` is used on the input.
- I renamed it to `triggerControlValidation` because there is another function called `triggerValidation` which is for internal control state service.

Reply:
Resolved!